src/js/modules/infragistics.ui.editors.js

```diff
@@ -12059,5 +12062,666 @@
+			} else {
+				minValue = 0;
+			}
+			if (maxValue) {
+				maxValue = maxValue.getHours() * 60 + maxValue.getMinutes();
+			} else {
+				maxValue = 1440;
+			}
+
+			var timeDeltaMinutes = this.options.itemsDelta.hours * 60 + this.options.itemsDelta.minutes;
+			var startMinutes = minValue / timeDeltaMinutes;
+			var dropDownItemsCount = 0;
+
+			if (timeDeltaMinutes > 0 && timeDeltaMinutes <= 1440) {
+				dropDownItemsCount = 1440 / timeDeltaMinutes;
+			}
+
+			var initDate = new Date();
+			initDate.setHours(0);
+			initDate.setMinutes(0);
+			initDate.setSeconds(0);
+
+			this.options.listItems = [];
+
+			for (var i = startMinutes; i < dropDownItemsCount; i++) {
+				var date = new Date(initDate);
+				date.setMinutes(timeDeltaMinutes * i);
+				if (timeDeltaMinutes * i >= minValue && timeDeltaMinutes * i <= maxValue) {
+					this.options.listItems.push(date);
+				}
+			}
+		},
+		_parseTimeMask: function(mask) {
+			return mask.replace(/M/g, "\\M").replace(/d/g, "\\d").replace(/y/g, "\\y").replace(/s/g, "\\s").replace(/f/g, "\\f");
+		},
+		_validateValue: function (val) { //igTimePicker
+			if (this._super(val)) {
+				var result;
+				if (val === undefined) {
+					result = false;
+				} else if (val === null) {
+					if (this.options.allowNullValue) {
+						result = val === this.options.nullValue ? true : false;
+					} else {
+						result = false;
+					}
+				} else if (this.options.isLimitedToListValues && this._dropDownList) {
+					if (this._valueIndexInList(val) !== -1) {
+						result = true;
+					} else {
+						this._sendNotification("warning", { optName: "allowedValuesMsg" });
+						result = false;
+					}
+				} else {
+					result = true;
+				}
+				return result;
+			}
+			return false;
+		},
+		_setOption: function (option, value) { //igTimePicker
+			var prevValue = this.options[ option ];
+			if (prevValue === value) {
+				return;
+			}
+
+			$.Widget.prototype._setOption.apply(this, arguments);
+			switch (option) {
+				case "timeDisplayFormat":
+					this._super("dateDisplayFormat", this._parseTimeMask(value));
+					break;
+				case "timeInputFormat":
+				case "itemsDelta":
+				case "minValue":
+				case "maxValue":
+					this.options[ option ] = prevValue;
+					throw new Error(this._getLocaleValue("setOptionError") + option);
+				case "value":
+					if ($.type(value) === "date") {
+						this.options[ option ] = prevValue;
+						this._super(option, value);
+					} else {
+						this._super(option, this._parseDateFromMaskedValue(value));
+					}
+					break;
+				default:
+
+				// In case no propery matches, we call the super. Into the base widget default statement breaks
+				this.options[ option ] = prevValue;
+				this._super(option, value);
+				break;
+			}
+		},
+		_updateValue: function (value) { //igTimePicker
+			if (value === null || $.type(value) === "date" || value === "") {
+				this._super(value);
+				if ($.type(value) === "date" && this._dropDownList !== undefined) {

```

Comment:
So if the value is null/empty (thus `$.type(value) === "date"` is false) the dropdown selection won't update?
I think it should always update if the value changes, so this entire block can be simplified with `_super` and `_updateDropdownSelection` after the value handling

Reply:
On a second thought, there shouldn't be additional `new Date(value)` parse, because depending on data mode your value could very well be a simple string that's not to be parsed at all.